src/methods/wallet-send-calls.test.ts

```diff
@@ -0,0 +1,176 @@
+  });
+
+  it('throws if missing properties', async () => {
+    params[0].from = undefined as never;
+    params[0].chainId = undefined as never;
+    params[0].calls = undefined as never;
+
+    await expect(callMethod()).rejects.toMatchInlineSnapshot(`
+            [Error: Invalid params
+
+            0 > from - Expected a string, but received: undefined
+            0 > calls - Expected an array value, but received: undefined]
+          `);
+  });
+
+  it('throws if wrong types', async () => {
+    params[0].from = '123' as never;
+    params[0].chainId = 123 as never;
+    params[0].calls = '123' as never;
+
+    await expect(callMethod()).rejects.toMatchInlineSnapshot(`
+            [Error: Invalid params
+
+            0 > from - Expected a string matching \`/^0x[0-9a-fA-F]{40}$/\` but received "123"
+            0 > chainId - Expected a string, but received: 123
+            0 > calls - Expected an array value, but received: "123"]
+          `);
+  });
+
+  it('throws if calls have wrong types', async () => {
+    params[0].calls[0].data = 123 as never;
+    params[0].calls[0].to = 123 as never;
+    params[0].calls[0].value = 123 as never;
+
+    await expect(callMethod()).rejects.toMatchInlineSnapshot(`
+            [Error: Invalid params
+
+            0 > calls > 0 > to - Expected a string, but received: 123
+            0 > calls > 0 > data - Expected a string, but received: 123
+            0 > calls > 0 > value - Expected a string, but received: 123]
+          `);
+  });
+
+  it('throws if not hex', async () => {
+    params[0].from = '123' as never;
+    params[0].chainId = '123' as never;
+    params[0].calls[0].data = '123' as never;
+    params[0].calls[0].to = '123' as never;
+    params[0].calls[0].value = '123' as never;
+
+    await expect(callMethod()).rejects.toMatchInlineSnapshot(`
+            [Error: Invalid params
+
+            0 > from - Expected a string matching \`/^0x[0-9a-fA-F]{40}$/\` but received "123"
+            0 > chainId - Expected a string matching \`/^0x[0-9a-f]+$/\` but received "123"
+            0 > calls > 0 > to - Expected a string matching \`/^0x[0-9a-fA-F]{40}$/\` but received "123"
+            0 > calls > 0 > data - Expected a string matching \`/^0x[0-9a-f]+$/\` but received "123"
+            0 > calls > 0 > value - Expected a string matching \`/^0x[0-9a-f]+$/\` but received "123"]
+          `);
+  });
+
+  it('throws if addresses are wrong length', async () => {
+    params[0].from = '0x123' as never;
+    params[0].calls[0].to = '0x123' as never;
+
+    await expect(callMethod()).rejects.toMatchInlineSnapshot(`
+            [Error: Invalid params
+
+            0 > from - Expected a string matching \`/^0x[0-9a-fA-F]{40}$/\` but received "0x123"
+            0 > calls > 0 > to - Expected a string matching \`/^0x[0-9a-fA-F]{40}$/\` but received "0x123"]
+          `);
+  });
+
+  it('throws if from is not in accounts', async () => {
+    getAccountsMock.mockResolvedValueOnce([]);
```

Comment:
should `validateAndNormalizeKeyholder` be mocked instead?

Reply:
A trade-off between test overlap and fragility. Since it was a one liner, I opted to treat the util as an implementation detail and not couple the unit tests to it.

Reply:
Fair point. I struggle with striking a balance in this area. Cool thanks!